CMakeLists.txt

```diff
@@ -192,8 +198,12 @@ else()
 endif()
 
 if(NOT WIN32)
-  core_optional_dep(OpenGl)
-  if(OPENGL_FOUND AND NOT APPLE)
+  if(NOT ROCKCHIP_FOUND)
+    core_optional_dep(OpenGl)
+  else()
+    core_optional_dep(XRandR ENABLE_X11)
+  endif()
+  if(OPENGL_FOUND AND NOT APPLE AND NOT ROCKCHIP_FOUND)
```

Comment:
This entire thing needs a redesign that makes it clear what modules are required for what platforms. X11, XrandR for example are no optional modules. Those are requirements for X11, that is a platform by itself. OpenGLES is not optional for Rockchip, it is a requirement. If you wnat to build for X11, Rockchip, or other platforms, build needs to fail if required modules are not found. Going an alternative branch makes no sense.